Tools/gittools/pre_commit_copyright.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+
+# CAUTION: The first docstring in this file will be the copyright notice that will be
+# looked for in all file paths that the pre-commit hook passes in.  If all files
+# contain the __doc__ text below, the commit will be allowed to proceed.
+
+"""ArduPilot.org.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+"""
+
+# --- Do not merges these two docstrings.  See CAUTION above. ---
+
+import sys
+from pathlib import Path
+
+r"""
+A pre-commit hook that will use the content of __doc__ and then ensure all files
+processed contain that same copyright information.  This is useful to ensure that all
+those files consistent copyright notices, which is important for legal and compliance
+reasons.  If any files does not contain the expected copyright information, the commit
+will be aborted and an error message will be displayed with the file paths of all files
+that do not match.
+An entry will be placed into the local `.pre-commit-config.yaml` file to run this job.
+```yaml
+  - repo: local
+    hooks:
+    -   id: check-copyright
+        name: check copyright notice in files
+        entry: Tools/gittools/pre_commit_copyright.py
+        language: python
+        files: |
+            (?x)^(
+              Tools/ros2/.*\.py
+            )$
+```
+
+Usage:
+1. Place this script in the `Tools/gittools/pre_commit_copyright.py` file.
+2. Ensure the docstring of this file is set to the expected copyright notice.
+3. Add the entry to your `.pre-commit-config.yaml` file as shown above.
+4. Run `pre-commit install` to set up the pre-commit hooks.
+5. Now, when you try to commit changes, this hook will check for the copyright notice.
+6. If any files do not contain the expected copyright notice, the commit will be aborted.
+7. The error message will list all files that do not match the expected copyright notice.
+"""
+
+exclude_file_paths = {
+    "Tools/ros2/ardupilot_dds_tests/setup.py",
+    "Tools/ros2/ardupilot_dds_tests/test/test_copyright.py",
+    "Tools/ros2/ardupilot_dds_tests/test/test_pep257.py",
+}
```

Comment:
Thoughts on putting these excludes in the pre-commit config instead like the other excludes?